metaflow/plugins/kfp/tests/flows/kfp_flow.py

```diff
@@ -27,22 +48,31 @@ def start(self):
         self.dividend = 26
         self.divisor = 7
         self.next(self.end)
+        env_image_tag = os.getenv("IMAGE_TAG")
+        assert not env_image_tag.endswith("_kfp_step")
+        assert_step_image(env_image_tag)
 
     @kfp(
         preceding_component=div_mod,
         preceding_component_inputs="dividend divisor",
         preceding_component_outputs="quotient remainder",
+        image=os.getenv("KFP_STEP_IMAGE"),
     )
     @step
     def end(self):
         """
         Validate that the results of the preceding div_mod KFP step are bound
-        to Metaflow state
+        to Metaflow state. We use os.getenv instead of environ to ensure this flow
+        runs correctly on the local machine (which doesn't have the KFP_STEP_IMAGE
+        and IMAGE_TAG env vars).
```

Comment:
I'm not able to parse this statement.  Why use "os.getenv instead of environ"?
Also, even on a local machine this test would fail because `IMAGE_TAG` doesn't exist, so why the comment?

Reply:
Removed this comment entirely, updated flow to use `is_on_kubernetes()` to ensure it works on local execution.